src/workspace-data/import-jobs.ts

```diff
@@ -35,6 +35,24 @@ export const useImportJobs = (workspace: WorkspaceWrapper): UseImportJobsResult
           .workspace(namespace, name)
           .listImportJobs(true);
 
+        // In cases where jobs failed before this query completes, users won't be notified of failure
+        // But they are momentarily in the job store, so check for them and notify here
+        const jobsToNotify = allRunningJobs.filter(
+          (job) => !runningJobsInWorkspace.some(({ jobId }) => job.jobId === jobId)
+        );
+
+        jobsToNotify.forEach((job) => {
+          Workspaces(signal)
+            .workspace(namespace, name)
+            .getImportJobStatus(job.jobId)
+            .then((fetchedJob) => {
+              if (fetchedJob.status === 'Error') {
+                notify('error', 'Error importing data.', { message: fetchedJob.message });
+                clearNotification(job.jobId);
+              }
+            });
+        });
+
```

Comment:
If I'm understanding this correctly, `jobsToNotify` will contain any/all jobs from other workspaces. This means we'll make an ajax call to ask for status of that job, but specifying this workspace's namespace/name, which is bound to fail - it'll be a 404 (I think; it'll be some kind of error).

Perhaps the `const jobsToNotify = allRunningJobs….` filter should include something about only checking for jobs in the current workspace?

Reply:
Good call!  Updated.